compiler/rustc_arena/src/lib.rs

```diff
@@ -655,6 +669,14 @@ pub macro declare_arena([$($a:tt $name:ident: $ty:ty,)*]) {
             self.dropless.alloc_slice(value)
         }
 
+        #[inline]
+        pub fn alloc_str(&self, string: &str) -> &str {
+            if string.is_empty() {
+                return "";
```

Comment:
Maybe remove this case and keep the panic?
No current use requires it.

Reply:
We can if you prefer, but I think for perf the branchiness should end up equivalent, just changing whether it jumps to a panic or return. And the empty check here is consistent with its `alloc_slice` neighbor. A panic still makes sense if we want that treated as a *bug* -- is that it?

Reply:
In `symbol.rs` that would certainly be a bug, with `SymbolName::new` I'm less sure, but it looks like the string cannot be empty there either.
I don't have much preference here, let's land as is.